src/components/LoadingOverlay/LoadingOverlay.tsx

```diff
@@ -0,0 +1,32 @@
+import { Loader2 } from 'lucide-react';
+import './LoadingOverlay.scss';
+
+interface LoadingOverlayProps {
+  isLoading: boolean;
+  message?: string;
+}
+
+export function LoadingOverlay({
+  isLoading,
+  message = 'Loading...',
+}: LoadingOverlayProps) {
+  if (!isLoading) return null;
+
+  return (
+    <section
+      className="loading-overlay"
+      role="alert"
+      aria-live="polite"
+      aria-busy="true"
+    >
+      <article className="loading-overlay__content">
```

Comment:
sam spinner nie ma znaczenia semantycznego - to po prostu jakiś wizualny ulepszać, który nie ma w sobie treści mającej jakieś znaczenie - tutaj zwykły div załatwi sprawę